system/src/androidLibMain/kotlin/com/splendo/kaluga/system/network/services/ConnectivityCallbackNetworkManager.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network.services
+
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.splendo.kaluga.system.network.NetworkHelper
+import com.splendo.kaluga.system.network.NetworkStateChange
+
+@RequiresApi(Build.VERSION_CODES.M)
```

Comment:
Are we doing anything for less than `M` actually?

Reply:
Yes, see ConnectivityReceiverNetworkManager

Reply:
I see it 👍